docs/enhancements/tenancy-based-auth.md

```diff
@@ -0,0 +1,103 @@
+- A tenancy-based read-only access to resources. 
+- Users should be allowed to grant editing capabilities to their resources to other users in their organization.
+
+## Goals
+The following tables detail the access permission rules for various resources:
+
+For clusters and cluster-based resources (all APIs that starts with `/v2/clusters/{cluster_id}`):
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| List Cluster   | Returns clusters owned by all members of the organization                                      |
+| Get  Cluster   | Allowed for all members of the organization                                                    |
+| Edit Cluster   | Allowed for the owner and for each user that has update access for this cluster's subscription |
+| Delete Cluster | Allowed for the owner and for each user that has remove access for this cluster's subscription |
+
+For InfraEnv and InfraEnv-based resources like hosts (all APIs that starts with `/v2/infra-envs/{infra_env_id`):
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| List InfraEnvs | Returns bound InfraEnvs belonging to the organization and unbound InfraEnvs owned by the caller                                                                                                            |
+| Get    InfraEnv| For bound InfraEnv, allowed for all members of the organization<br>For unbound InfraEnv allowed for the owner                                                                                                         |
+| Edit InfraEnv  | For bound InfraEnv, allowed for the owner and for each user that has update access<br>for the bound cluster's subscription. For unbound InfraEnv, allowed for the owner                                               |
+| Delete InfraEnv| For bound InfraEnv, allowed for the owner and for each user that has remove access<br>for the bound cluster's subscription. For unbound InfraEnv, allowed for the owner                                               |
+
+For day2 cluster operations:
+| Action         | Authorization Access                                                                           |
+|----------------|------------------------------------------------------------------------------------------------|
+| Add Host       | Allowed for every user who has update access for the day1's cluster subscription               |
+
+Filtering for you own clusters:
+The list-clusters API shall have an additional query parameter that filters the clusters by owners only:
+`GET /v2/clusters?owneronly=true`
+
+## Authorization Flow Overview
```

Comment:
Okay after reading the entire section I realized it's only describing the *current* state of things and not part of the enhancement itself. Is that correct? If so it would be nice to emphasize that this is only an explanation of the current state of things and not an enhancement